src/core/control/tools/EditSelection.cpp

```diff
@@ -1038,6 +1084,74 @@ void EditSelection::drawDeleteRect(cairo_t* cr, double x, double y, double zoom)
     cairo_stroke(cr);
 }
 
+/**
+ * draws an indicator where you can cut the selection
+ */
+void EditSelection::drawAnchorCut(cairo_t* cr, double x, double y, double zoom) const {
+    cairo_set_source_rgb(cr, 0, 0, 0);
+    cairo_rectangle(cr, x * zoom - (this->btnWidth / 2), y * zoom - (this->btnWidth / 2), this->btnWidth,
+                    this->btnWidth);
+    cairo_stroke(cr);
+    cairo_set_source_rgb(cr, 1, 0, 0);
+    cairo_move_to(cr, x * zoom - (this->btnWidth / 2), y * zoom - (this->btnWidth / 2));
```

Comment:
Since we'll be making the style consistent, the "setup" logic could be extracted into a helper function:

```c++
namespace {
void prepareDrawAnchor(cairo_t* cr, double x, double y, double zoom, double btnWidth) {
    cairo_set_source_rgb(cr, 0, 0, 0);
    cairo_rectangle(cr, x * zoom - (btnWidth / 2), y * zoom - (btnWidth / 2), btnWidth,
                    btnWidth);
    cairo_stroke(cr);
    cairo_set_source_rgb(cr, 1, 0, 0);
    cairo_move_to(cr, x * zoom - (btnWidth / 2), y * zoom - (btnWidth / 2));
}
} // namespace
```

Reply:
If you want to refactor, I'd suggest to refactor the existing code first, then rebase/recode anything new on top. That way we know which (possibly unexpected) changes in behaviour come from which code changes.

Reply:
Feel free to refactor within a separate commit that is then part of this PR.